DPL: more fixes to handle the ROOT v6-36-00

ktf opened the pull request:
- More tweaks to get ROOT::Experimental out of the way
- RNTuples cannot have / as first character in name

ktf commented:
@pzhristov this should solve the remaining issues on my side for the new ROOT.

pzhristov commented:
@ktf It seems we still have the problem in framework:root. Do we need PR #14332 for the successful completion of the tests?

ktf commented:
I fear some of my changes for the new ROOT broke the old one. Checking.

ktf commented:
https://github.com/AliceO2Group/AliceO2/pull/14332 is unrelated. Originally, it happened to crash in the same place, which is why I was originally confused, and it should now be good to go.

pzhristov commented:
@ktf I confirm that this PR fixes the issue of framework:root with Root v6-36-00.

ktf committed:
DPL: more fixes to handle the ROOT v6-36-00

- More tweaks to get ROOT::Experimental out of the way
- RNTuples cannot have / as first character in name

ktf commented:
Code not used / unchanged until we switch to  ROOT v6-36-00.

ktf commented:
Also tested on hyperloop.
